kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepository.java

```diff
@@ -106,10 +114,16 @@ public synchronized void store(@NotNull String serverId, @NotNull KieServerState
             DeploymentConfig dc = client.deploymentConfigs().withName(serverId).get();
             ConfigMap cm = client.configMaps().withName(serverId).get();
             String stateXML = xs.toXML(kieServerState);
-            if (cm == null) {
-                throw new IllegalStateException("KieServerState ConfigMap must exist before update.");
-            } else {
+            
+            if (Boolean.parseBoolean(kieServerState.getConfiguration()
```

Comment:
Does this mean that for immutable Kie server a new CM is created/overridden but no rollout is triggered, for other cases data is updated and rollout is triggered, and for the last case - CM doesn't contain USED or IMMUTABLE labels or container deploying or undeploying is done in IMMUTABLE CM - nothing is done?
Do I understand this correctly?

Reply:
Once the immutable kie server CM has been created by the S2I/KieServerFileInit process, any update to it is not allowed and blocked.